### ISSUE TITLE
Clicking in outline has no effect

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#136692]
Issue Type: <b>Bug</b>

1. Open this notebook in VS Code: https://www.kaggle.com/alexisbcook/titanic-tutorial

2. Open the outline view and try clicking on any of the entries

**bug**
Nothing happens

Extension version: 2021.10.1001414422
VS Code version: Code - Insiders 1.63.0-insider (Universal) (552af97a0170b43c9323bec62211feb184da9749, 2021-11-05T05:15:13.016Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (0.11GB free)|
|Process Argv|--crash-reporter-id 48781ca2-1705-4f64-9bab-325055aab55d|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
vsccppwt:30382699
bridge0708:30335490
pygetstartedt2:30353727
bridge0723:30353136
pythonrunftest32:30365365
pythonf5test824:30361779
javagetstartedt:30350119
pythonvspyt187:30365360
pydsgst2:30361790
vsaa593cf:30376535
vsc1dsc:30394499

```

</details>

**Comment by jrieken** [comment]
The outline properly calls [openEditor](https://github.com/microsoft/vscode/blob/5c8ba90f12af6cc02ceb40989b400ce77003fd1e/src/vs/workbench/contrib/notebook/browser/contrib/outline/notebookOutline.ts#L544) but this code is never executed (which is vital because it transforms the cell-uri to cell options): 

https://github.com/microsoft/vscode/blob/8ad817b486cd4311bd05d33e718808293cfe2c9a/src/vs/workbench/contrib/notebook/browser/notebookServiceImpl.ts#L168

**Comment by bpasero** [comment]
Fyi https://github.com/microsoft/vscode/commit/6ed67dd61ebbd218892b3c47008cac33da0ac899 changes `openEditor` to return `undefined` if that operation was canceled by another operation.

@lramos15 not sure if this change is maybe triggering an issue with the editor resolver service? Though I would somewhat doubt that because this happens very late at the point where the resolved input is given to the editor pane.

**Comment by lramos15** [comment]
@bpasero The resolver returns an editor input for the notebook when the outline is clicked. I also see that the operation is not marked as changed or cancelled on the pane.

**Comment by bpasero** [comment]
Optimistically closing given https://github.com/microsoft/vscode/commit/42f10bb643c63c91421a62e6fa8d08d5297dd09f, but please confirm.

**Comment by jrieken** [comment]
Still reproduces when running out of sources

**Comment by bpasero** [comment]
True, this is unrelated to my change around `openEditors`, should have tested this before closing.

**Comment by bpasero** [comment]
@rebornix from https://github.com/microsoft/vscode/commit/ec2b75ac91c46af169d88c1c3bdc2c807cbf7424

I urge you do add a test case for this, please.

**Comment by rebornix** [comment]
@bpasero sorry about that, I should add some tests but firstly I'd like to know what to cover and what type of test we are going to add, since it's a workbench-notebook integration.

ec2b75a was added to fix https://github.com/microsoft/vscode/issues/133761, in which the editor service wish notebook cell input to match notebook editor input. Since we change the behavior of this `EditorInput.matches` override, it affects how `editorResolverService` works in another place, in terms of if the editor options we passed in are respected or not.

When one click on the outline, we will use the editor service to open the cell, with a custom editor options.

https://github.com/microsoft/vscode/blob/5c8ba90f12af6cc02ceb40989b400ce77003fd1e/src/vs/workbench/contrib/notebook/browser/contrib/outline/notebookOutline.ts#L543-L548

Then in `editorResolverService#resolveEditor`, it checks if the notebook cell matches the notebook editor input or not.

https://github.com/microsoft/vscode/blob/419744928a3cb7cc5207ca46a6aef75bc0ac39b8/src/vs/workbench/services/editor/browser/editorResolverService.ts#L207-L225

if they match, which is line 211, it will open the `activeEditor`, with untypeEditor's default options. If they don't match, which will go to line 224, `...input` ensure that our custom editor options are carried over (`cellOptions` is the property we care about). 

---

Long story short, the return result of `EditorInput.matches` for embed editor inputs (NotebookInput and EditorInput for cells) can affect behaviors of editor resolver service in different ways in different places. It sounds like `matches` is overloaded here, now I'm not sure what it should return to be 100% correct. Reverted my previous commit to unblock Outline.

**Comment by rebornix** [comment]
After discussing with @lramos15 we knew how Outline worked before:

* Outline tries to open cell uri
* It doesn't match active editor
* We call `doResolveEditor`, Notebook contribution creates a NotebookEditorInput with custom options
* The returned NotebookEditorInput matches the active editor

**Comment by lramos15** [comment]
This is fixed with @rebornix's revert. Will need to discuss better solution for #133761

**Comment by bpasero** [comment]
@rebornix 

> I should add some tests but firstly I'd like to know what to cover and what type of test we are going to add, since it's a workbench-notebook integration

How about a test that first opens a notebook and then opens a cell-uri verifying that this changed the scroll position of the notebook?
